addons/website/static/src/js/editor/widget_link.js

```diff
@@ -66,7 +67,14 @@ weWidgets.LinkDialog.include({
 
         function always() {
             $anchorsLoading.addClass('d-none');
-            $selectMenu.prop("selectedIndex", -1);
+            const anchor = `#${$selectMenu.data("anchor-for").split('#')[1]}`;
+            let anchorIndex = -1;
+            if (anchor) {
+                const optionEls = $selectMenu[0].querySelectorAll('option');
+                const anchorList = Array.from(optionEls).map(option => option.textContent);
+                anchorIndex = anchorList.indexOf(anchor);
```

Comment:
This builds the list of text of all options then goes through the list to find the index. This should be written like this (untested):
```javascript
const optionEls = $selectMenu[0].querySelectorAll('option');
anchorIndex = Array.from(optionEls).findIndex(el => el.textContent === anchor);
```

That way you stop looping through the option as soon as you found the right value. It's alos shorter to write so :)